app/src/main/java/woowacourse/payments/component/NumberTextField.kt

```diff
@@ -0,0 +1,71 @@
+package woowacourse.payments.component
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import woowacourse.payments.InputType
+import woowacourse.payments.R
+import woowacourse.payments.ui.theme.Gray100
+import woowacourse.payments.ui.theme.Gray200
+
+@Composable
+fun NumberTextField(
+    modifier: Modifier = Modifier,
+    label: Int,
+    placeholder: Int,
+    inputType: InputType,
+    value: String,
+    onValueChange: (String) -> Unit,
+) {
+    val textFieldValue =
+        remember(value) {
+            TextFieldValue(text = value, selection = TextRange(value.length))
+        }
+
+    OutlinedTextField(
+        value = textFieldValue,
+        onValueChange = { input ->
+            val formatted = inputType.format(input.text)
+            onValueChange(formatted)
+        },
+        label = { Text(stringResource(label), color = Gray200) },
+        placeholder = { Text(stringResource(placeholder), color = Gray100) },
+        visualTransformation =
+            if (inputType is InputType.Password) {
+                PasswordVisualTransformation()
+            } else {
+                VisualTransformation.None
+            },
+        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.NumberPassword),
+        modifier =
+            modifier.then(
+                Modifier
+                    .padding(bottom = 30.dp)
+                    .padding(horizontal = 24.dp),
+            ),
```

Comment:
```suggestion
        modifier =
            modifier
                .padding(bottom = 30.dp)
                .padding(horizontal = 24.dp),
```
여기서 then을 쓰셨는데 위와 같이 변경을 하게 되어도 같은 동작을 하는 거 같은데 어떤 차이가 있나요?🤔

Reply:
두 코드의 동작상 차이는 없는 것 같네요 😅

저는 처음에 코드를 작성할 때 제가 외부에서 설정할 Modifier에 추가 설정을 해주려면 `then`을 꼭 붙여야하는줄 알고 붙였었어요 🥲
현재 코드에서 두 코드의 차이는 없지만, 만약 **분기처리를 해서 조건에 따라 Modifier를 설정**해주고 싶다면 `then`을 사용해서 다음과 같이 사용할 수 있다고 합니당

```kotlin
var enabled = true
modifier = modifier.then(
    if (enabled) Modifier.padding(16.dp) else Modifier
)
```

[수정했습니다!](https://github.com/woowacourse/android-payments/pull/13/commits/63b586f3a52f92ac230cf4f5dd573ed0741c9c3f)